File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3c4ev3/text_colours_messages_functions.cpp
#define PROGRAM_NAME "NEPOprog"
#define WHEEL_DIAMETER 5.6
#define TRACK_WIDTH 18.0

#include <ev3.h>
#include <math.h>
#include <list>
#include "NEPODefs.h"


void ____text();
void ____colours();
void ____messages();
void ____function_parameter(double ___x, bool ___x2, std::string ___x3, Color ___x4, BluetoothConnectionHandle ___x5, std::list<double> ___x6, std::list<bool> ___x7, std::list<std::string> ___x8, std::list<Color> ___x9, std::list<BluetoothConnectionHandle> ___x10);
double ____function_return_numberVar();
bool ____function_return_booleanVar();
std::string ____function_return_stringVar();
Color ____function_return_colourVar();
BluetoothConnectionHandle ____function_return_connectionVar();
std::list<double> ____function_return_numberList();
std::list<bool> ____function_return_booleanList();
std::list<std::string> ____function_return_stringList();
std::list<Color> ____function_return_colourList();
std::list<BluetoothConnectionHandle> ____function_return_connectionList();

double ___numberVar = 0;
bool ___booleanVar = true;
std::string ___stringVar = "";
Color ___colourVar = White;
BluetoothConnectionHandle ___connectionVar = NULL;
std::list<double> ___numberList = ((std::list<double>){0, 0});
std::list<bool> ___booleanList = ((std::list<bool>){true, true});
std::list<std::string> ___stringList = ((std::list<std::string>){"", ""});
std::list<Color> ___colourList = ((std::list<Color>){White, White});
std::list<BluetoothConnectionHandle> ___connectionList = ((std::list<BluetoothConnectionHandle>){___connectionVar, ___connectionVar});

void ____text() {
    DrawString(ToString(""), ___numberVar, ___numberVar);
    //
    DrawString(ToString(ToString(___stringVar) + ToString(___stringVar)), ___numberVar, ___numberVar);
    ___stringVar += ___stringVar;
}

void ____colours() {
    DrawString(ToString(Black), ___numberVar, ___numberVar);
    DrawString(ToString(Blue), ___numberVar, ___numberVar);
    DrawString(ToString(Green), ___numberVar, ___numberVar);
    DrawString(ToString(Brown), ___numberVar, ___numberVar);
    DrawString(ToString(None), ___numberVar, ___numberVar);
    DrawString(ToString(Red), ___numberVar, ___numberVar);
    DrawString(ToString(Yellow), ___numberVar, ___numberVar);
    DrawString(ToString(White), ___numberVar, ___numberVar);
}

void ____messages() {
    DrawString(ToString(NEPOConnectTo(___stringVar)), ___numberVar, ___numberVar);
    NEPOSendStringTo(___connectionVar, ___stringVar);
    DrawString(ToString(NEPOReceiveStringFrom(___connectionVar)), ___numberVar, ___numberVar);
    DrawString(ToString(NEPOWaitConnection()), ___numberVar, ___numberVar);
}

void ____function_parameter(double ___x, bool ___x2, std::string ___x3, Color ___x4, BluetoothConnectionHandle ___x5, std::list<double> ___x6, std::list<bool> ___x7, std::list<std::string> ___x8, std::list<Color> ___x9, std::list<BluetoothConnectionHandle> ___x10) {
    if (___booleanVar) return ;
}

double ____function_return_numberVar() {
    return ___numberVar;
}

bool ____function_return_booleanVar() {
    return ___booleanVar;
}

std::string ____function_return_stringVar() {
    return ___stringVar;
}

Color ____function_return_colourVar() {
    return ___colourVar;
}

BluetoothConnectionHandle ____function_return_connectionVar() {
    return ___connectionVar;
}

std::list<double> ____function_return_numberList() {
    return ___numberList;
}

std::list<bool> ____function_return_booleanList() {
    return ___booleanList;
}

std::list<std::string> ____function_return_stringList() {
    return ___stringList;
}

std::list<Color> ____function_return_colourList() {
    return ___colourList;
}

std::list<BluetoothConnectionHandle> ____function_return_connectionList() {
    return ___connectionList;
}

int main () {
    NEPOInitEV3();
    NEPOSetAllSensors(NULL, NULL, NULL, NULL);
    startLoggingThread(OUT_BC);


    ____text();
    ____colours();
    ____messages();
    ____function_parameter(___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList);
    DrawString(ToString(____function_return_numberVar()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_booleanVar()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_stringVar()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_colourVar()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_connectionVar()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_numberList()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_booleanList()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_stringList()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_colourList()), ___numberVar, ___numberVar);
    DrawString(ToString(____function_return_connectionList()), ___numberVar, ___numberVar);

    NEPOFreeEV3();
    return 0;
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3dev/text_colours_messages_functions.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
    },
}
hal = Hal(_brickConfiguration)

___numberVar = 0
___booleanVar = True
___stringVar = ""
___colourVar = 'white'
___connectionVar = None
___numberList = [0, 0]
___booleanList = [True, True]
___stringList = ["", ""]
___colourList = ['white', 'white']
___connectionList = [___connectionVar, ___connectionVar]
def ____text():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    hal.drawText("", ___numberVar, ___numberVar)
    #
    hal.drawText(str("".join(str(arg) for arg in [___stringVar, ___stringVar])), ___numberVar, ___numberVar)
    ___stringVar += str(___stringVar)

def ____colours():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    hal.drawText(str('black'), ___numberVar, ___numberVar)
    hal.drawText(str('blue'), ___numberVar, ___numberVar)
    hal.drawText(str('green'), ___numberVar, ___numberVar)
    hal.drawText(str('brown'), ___numberVar, ___numberVar)
    hal.drawText(str('none'), ___numberVar, ___numberVar)
    hal.drawText(str('red'), ___numberVar, ___numberVar)
    hal.drawText(str('yellow'), ___numberVar, ___numberVar)
    hal.drawText(str('white'), ___numberVar, ___numberVar)

def ____messages():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    hal.drawText(str(hal.establishConnectionTo(str(___stringVar))), ___numberVar, ___numberVar)
    hal.sendMessage(___connectionVar, str(___stringVar))
    hal.drawText(str(hal.readMessage(___connectionVar)), ___numberVar, ___numberVar)
    hal.drawText(str(hal.waitForConnection()), ___numberVar, ___numberVar)

def ____function_parameter(___x, ___x2, ___x3, ___x4, ___x5, ___x6, ___x7, ___x8, ___x9, ___x10):
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    if ___booleanVar: return None

def ____function_return_numberVar():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___numberVar

def ____function_return_booleanVar():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___booleanVar

def ____function_return_stringVar():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___stringVar

def ____function_return_colourVar():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___colourVar

def ____function_return_connectionVar():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___connectionVar

def ____function_return_numberList():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___numberList

def ____function_return_booleanList():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___booleanList

def ____function_return_stringList():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___stringList

def ____function_return_colourList():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___colourList

def ____function_return_connectionList():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    return ___connectionList

def run():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    ____text()
    ____colours()
    ____messages()
    ____function_parameter(___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList)
    hal.drawText(str(____function_return_numberVar()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_booleanVar()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_stringVar()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_colourVar()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_connectionVar()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_numberList()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_booleanList()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_stringList()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_colourList()), ___numberVar, ___numberVar)
    hal.drawText(str(____function_return_connectionList()), ___numberVar, ___numberVar)

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()